askama_shared/src/lib.rs

```diff
@@ -37,9 +37,21 @@ pub trait Template: fmt::Display {
         Ok(buf)
     }
 
-    /// Renders the template to the given `writer` buffer
+    /// Helper method which allocates a new `Vec<u8>` and renders into it
+    #[inline]
+    fn render_bytes(&self) -> Result<Vec<u8>> {
+        Ok(self.render()?.into_bytes())
+    }
+
+    /// Renders the template to the given `writer` fmt buffer
     fn render_into(&self, writer: &mut (impl std::fmt::Write + ?Sized)) -> Result<()>;
 
+    /// Renders the template to the given `writer` io buffer
+    #[inline]
+    fn render_into_io(&self, writer: &mut (impl std::io::Write + ?Sized)) -> std::io::Result<()> {
```

Comment:
This one seems reasonable, though the name seems a little unwieldy. Maybe `write_into()`? That's technically somewhat ambitious between `impl std::fmt::Write` and `impl std::io::Write` but that doesn't seem like a big deal.

Reply:
Renamed.